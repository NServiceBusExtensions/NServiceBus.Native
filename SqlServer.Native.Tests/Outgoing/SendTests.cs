using System;
using System.Collections.Generic;
using System.IO;
using System.Text;
using NServiceBus.Transport.SqlServerNative;
using ObjectApproval;
using Xunit;
using Xunit.Abstractions;

public class SendTests : TestBase
{
    static DateTime dateTime = new DateTime(2000, 1, 1, 1, 1, 1, DateTimeKind.Utc);

    string table = "SendTests";

    [Fact]
    public void Single_bytes()
    {
        var message = BuildBytesMessage("00000000-0000-0000-0000-000000000001");
        Send(message);
        ObjectApprover.VerifyWithJson(SqlHelper.ReadData(table));
    }

    [Fact]
    public void Single_bytes_nulls()
    {
        var sender = new Sender("SendTests");

        var message = BuildBytesNullMessage("00000000-0000-0000-0000-000000000001");
        sender.Send(Connection.ConnectionString, message).Await();
        ObjectApprover.VerifyWithJson(SqlHelper.ReadData(table));
    }

    [Fact]
    public void Single_stream()
    {
        var message = BuildStreamMessage("00000000-0000-0000-0000-000000000001");
        Send(message);
        ObjectApprover.VerifyWithJson(SqlHelper.ReadData(table));
    }

    [Fact]
    public void Single_stream_nulls()
    {
        var message = BuildStreamMessage("00000000-0000-0000-0000-000000000001");
        Send(message);
        ObjectApprover.VerifyWithJson(SqlHelper.ReadData(table));
    }

    [Fact]
    public void Batch()
    {
        var messages = new List<OutgoingMessage>
        {
            BuildBytesMessage("00000000-0000-0000-0000-000000000001"),
            BuildStreamMessage("00000000-0000-0000-0000-000000000002")
        };
        Send(messages);
        ObjectApprover.VerifyWithJson(SqlHelper.ReadData(table));
    }

    [Fact]
    public void Batch_nulls()
    {
        var messages = new List<OutgoingMessage>
        {
            BuildBytesNullMessage("00000000-0000-0000-0000-000000000001"),
            BuildStreamNullMessage("00000000-0000-0000-0000-000000000002")
        };
        Send(messages);
        ObjectApprover.VerifyWithJson(SqlHelper.ReadData(table));
    }

    void Send(List<OutgoingMessage> messages)
    {
        var sender = new Sender(table);
        sender.Send(Connection.ConnectionString, messages).Await();
    }

    void Send(OutgoingMessage message)
    {
        var sender = new Sender("SendTests");
        sender.Send(Connection.ConnectionString, message).Await();
    }

    static OutgoingMessage BuildBytesMessage(string guid)
    {
        return new OutgoingMessage(new Guid(guid), "theCorrelationId", "theReplyToAddress", dateTime, "headers", Encoding.UTF8.GetBytes("{}"));
    }

    static OutgoingMessage BuildStreamMessage(string guid)
    {
        var stream = new MemoryStream(Encoding.UTF8.GetBytes("{}"));
        return new OutgoingMessage(new Guid(guid), "theCorrelationId", "theReplyToAddress", dateTime, "headers", stream);
    }

    static OutgoingMessage BuildStreamNullMessage(string guid)
    {
        return new OutgoingMessage(new Guid(guid), bodyStream: null);
    }

    static OutgoingMessage BuildBytesNullMessage(string guid)
    {
        return new OutgoingMessage(new Guid(guid), bodyBytes: null);
    }

    public SendTests(ITestOutputHelper output) : base(output)
    {
        SqlHelpers.Drop(Connection.ConnectionString, table).Await();
        QueueCreator.Create(Connection.ConnectionString, table).Await();
    }
}