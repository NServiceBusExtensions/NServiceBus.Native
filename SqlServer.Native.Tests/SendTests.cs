using System;
using System.Collections.Generic;
using System.Text;
using ObjectApproval;
using SqlServer.Native;
using Xunit;

public class SendTests
{
    static DateTime dateTime = new DateTime(2000, 1, 1, 1, 1, 1, DateTimeKind.Utc);

    static SendTests()
    {
        DbSetup.Setup();
    }

    [Fact]
    public void SendSingle()
    {
        SqlHelpers.Drop(Connection.ConnectionString, "SendTests").Await();
        QueueCreator.Create(Connection.ConnectionString, "SendTests").Await();
        var sender = new Sender();

        var message = BuildMessage("00000000-0000-0000-0000-000000000001");
        sender.Send(Connection.ConnectionString, "SendTests", message).Await();
        ObjectApprover.VerifyWithJson(SqlHelper.ReadData("SendTests"));
    }

    [Fact]
    public void SendBatch()
    {
        SqlHelpers.Drop(Connection.ConnectionString, "SendTests").Await();
        QueueCreator.Create(Connection.ConnectionString, "SendTests").Await();
        var sender = new Sender();

        sender.Send(
            Connection.ConnectionString, "SendTests",
            new List<Message>
            {
                BuildMessage("00000000-0000-0000-0000-000000000001"),
                BuildMessage("00000000-0000-0000-0000-000000000002")
            }).Await();
        ObjectApprover.VerifyWithJson(SqlHelper.ReadData("SendTests"));
    }

    static Message BuildMessage(string guid)
    {
        return new Message(new Guid(guid), "theCorrelationId", "theReplyToAddress", dateTime, "headers", Encoding.UTF8.GetBytes("{}"));
    }
}