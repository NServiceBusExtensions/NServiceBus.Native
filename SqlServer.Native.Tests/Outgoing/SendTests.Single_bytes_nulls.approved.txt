[
  {
    "Id": "00000000-0000-0000-0000-000000000001",
    "CorrelationId": null,
    "ReplyToAddress": null,
    "Recoverable": true,
    "Expires": null,
    "Headers": "{}",
    "BodyString": null,
    "Body": null,
    "RowVersion": 1
  }
]